heed-traits/src/lib.rs

```diff
@@ -11,3 +12,44 @@ pub trait BytesDecode<'a> {
 
     fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem>;
 }
+
+impl<'a, A, B> BytesEncode<'a> for (A, B)
+where
+    A: BytesEncode<'a>,
+    A::EItem: Sized,
+    B: BytesEncode<'a>,
+{
+    type EItem = (A::EItem, B::EItem);
+
+    fn bytes_encode(item: &'a Self::EItem) -> Option<Cow<[u8]>> {
+        let first_bytes = A::bytes_encode(&item.0)?;
+        let second_bytes = B::bytes_encode(&item.1)?;
+        let mut buffer = Vec::with_capacity(
+            mem::size_of::<usize>()
+            + &first_bytes.len()
+            + &second_bytes.len());
+        // signify the length of the first item
+        buffer.extend_from_slice(&first_bytes.len().to_be_bytes());
```

Comment:
On 32bits platforms this is an u32 and on 64bits platforms this is an u64, it is not consistent, you should not do that but rather cast it into an u64.